lib/src/components/NxSearchDropdown/__tests__/NxSearchDropdown.test.tsx

```diff
@@ -4,225 +4,203 @@
-
-  afterEach(function() {
-    mountPoint.remove();
-  });
+      renderEl = rtlRenderElement<PropsWithRef>(NxSearchDropdown, minimalProps),
+      quickRender = rtlRender<PropsWithRef>(NxSearchDropdown, minimalProps);
 
-  it('renders a div with the nx-search-dropdown class', function() {
-    expect(getShallow()).toMatchSelector('div.nx-search-dropdown');
-  });
+  it('adds additional specified classnames', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' });
 
-  it('adds additional specified classnames to the div', function() {
-    const component = getShallow({ className: 'foo' });
+    expect(customEl).toHaveClass('foo');
 
-    expect(component).toHaveClassName('foo');
-    expect(component).toHaveClassName('nx-search-dropdown');
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
-  it('adds additional specified attributes to the div', function() {
-    const component = getShallow({ id: 'foo', title: 'bar' });
+  it('adds additional specified attributes', function() {
+    const el = renderEl({ id: 'foo', title: 'bar' });
 
-    expect(component).toHaveProp('id', 'foo');
-    expect(component).toHaveProp('title', 'bar');
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('title', 'bar');
   });
 
-  it('adds a specified ref to the div', function() {
-    const ref = React.createRef<HTMLDivElement>();
-    const component = mount(<><NxSearchDropdown ref={ref} { ...minimalProps } /></>);
+  it('adds a specified ref', function() {
+    const ref = React.createRef<HTMLDivElement>(),
+        renderedEl = renderEl({ ref });
 
-    expect(ref.current).toBeDefined();
-    expect(component.getDOMNode()).toBe(ref.current);
+    expect(ref.current).toBe(renderedEl);
   });
 
-  it('has an NxFilterInput child with the nx-search-dropdown__input class and searchbox role', function() {
-    const component = getMounted(),
-        filterInput = component.find(NxFilterInput),
-        nativeInput = filterInput.find('input');
+  it('has an input field with role="searchbox"', function() {
+    const el = quickRender(),
+        input = el.getByRole('searchbox');
 
-    expect(filterInput).toExist();
-    expect(filterInput).toHaveClassName('nx-search-dropdown__input');
-    expect(nativeInput).toHaveProp('role', 'searchbox');
+    expect(input).toBeInTheDocument();
   });
 
   it('sets the searchText as the value of the input', function() {
-    const component = getShallow({ searchText: 'foo' }),
-        input = component.find(NxFilterInput);
+    const el = quickRender({ searchText: 'foo' }),
+        input = el.getByRole('searchbox');
 
-    expect(input).toHaveProp('value', 'foo');
+    expect(input).toHaveAttribute('value', 'foo');
   });
 
-  it('calls onSearchTextChange whenver the input\'s onChange event fires', function() {
+  it('calls onSearchTextChange whenver the input\'s onChange event fires', async function() {
```

Comment:
This should be reworded in terms of user-facing events e.g. whenever the user types in the searchbox (and then other tests for other was the searchbox's value could change such as pressing ESC or clicking the clear button). Accordingly the test should use `userEvent` not `fireEvent`.

Reply:
> This should be reworded in terms of user-facing events e.g. whenever the user types in the searchbox (and then other tests for other was the searchbox's value could change such as pressing ESC or clicking the clear button). Accordingly the test should use `userEvent` not `fireEvent`.

I've re-worded tests so that they are user-facing events as suggested: cb95291101d94947a0c5dc2843368a1dad4501bd

Reply:
> I've re-worded tests so that they are user-facing events as suggested: https://github.com/sonatype/sonatype-react-shared-components/commit/cb95291101d94947a0c5dc2843368a1dad4501bd

I notice you removed mention of the props function names (e.g. `onSearch` as well).  That is _not_ what I meant.  The component's props API is one external surface.  The user is another.  The goal of these tests is to make sure that when the user does something user-y that the props API gets triggered accordingly.  So the relevant props API parts should still be in the test names.

Reply:
> > I've re-worded tests so that they are user-facing events as suggested: [cb95291](https://github.com/sonatype/sonatype-react-shared-components/commit/cb95291101d94947a0c5dc2843368a1dad4501bd)
> 
> I notice you removed mention of the props function names (e.g. `onSearch`) as well. That is _not_ what I meant. The component's props API is one external surface. The user is another. The goal of these tests is to make sure that when the user does something user-y that the props API gets triggered accordingly. So the relevant props API parts should still be in the test names.

I see what you mean. That makes sense. Added back relevant keywords: e25c45117205cd3c01d45f0a3adb3263833a1a70